packages/realm-react/src/index.tsx

```diff
@@ -135,11 +54,51 @@ type RealmContext = {
  * @param realmConfig - {@link Realm.Configuration} used to open the Realm
  * @returns An object containing a `RealmProvider` component, and `useRealm`, `useQuery` and `useObject` hooks
  */
-export const createRealmContext: (realmConfig?: Realm.Configuration) => RealmContext = (
-  realmConfig: Realm.Configuration = {},
-) => {
-  const RealmContext = createContext<Realm | null>(null);
-  const RealmProvider = createRealmProvider(realmConfig, RealmContext);
+export function createRealmContext(realmConfig: Realm.Configuration): RealmContext<RealmProviderFromConfiguration>;
+/**
+ * Creates Realm React hooks and Provider component for a given Realm instance.
+ * @example
+ * ```
+ * const realm = await Realm.open({ path: "example.realm" });
+ * const {useRealm, useQuery, useObject, RealmProvider} = createRealmContext(realm);
+ * ```
+ * @param realm - {@link Realm} instance
+ * @returns An object containing a `RealmProvider` component, and `useRealm`, `useQuery` and `useObject` hooks
+ */
+export function createRealmContext(realm: Realm): RealmContext<RealmProviderFromRealmFC>;
+/**
+ * Creates Realm React hooks and Provider component.
+ * @example
+ * ```
+ * class Task extends Realm.Object {
+ *  ...
+ *  static schema: ObjectSchema = {
+ *    name: 'Task',
+ *    primaryKey: '_id',
+ *    properties: {
+ *      ...
+ *    },
+ *  };
+ * }
+ * const {useRealm, useQuery, useObject, RealmProvider} = createRealmContext();
+ * ...
+ * <RealmProvider schema={[Task]}></RealmProvider>
+ * ```
+ * @example
+ * ```
+ * const realm = await Realm.open({ path: "example.realm", schema: [Task] });
+ * const {RealmProvider} = createRealmContext();
+ * ...
+ * <RealmProvider realm={realm}></RealmProvider>
+ * ```
+ * @returns An object containing a `RealmProvider` component, and `useRealm`, `useQuery` and `useObject` hooks
+ */
+export function createRealmContext(): RealmContext<DynamicRealmProvider>;
+export function createRealmContext(
+  realmOrConfig?: Realm | Realm.Configuration,
+): RealmContext<RealmProviderFromConfiguration | RealmProviderFromRealmFC | DynamicRealmProvider> {
+  const RealmContext = createContext<Realm | null>(realmOrConfig instanceof Realm ? realmOrConfig : null);
```

Comment:
A side-effect of passing the `Realm` instance as the `defaultValue` argument of `createContext` is that users providing the Realm via `createRealmContext` actually doesn't have to wrap consumers (including the hooks derived from the context) in a `RealmProvider`. This is a great feature and I think it should be documented (and tested if it isn't already).

Reply:
that's a great point. should we even return a RealmProvider in this case then? seems misleading unless we plan to add other props to this in the future